File: SDKs/googleapis.com/youtube/v3/java/lib/src/main/java/openapisdk/models/shared/ChannelSectionSnippet.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * ChannelSectionSnippet
 * Basic details about a channel section, including title, style and position.
**/
public class ChannelSectionSnippet {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("channelId")
    public String channelId;
    public ChannelSectionSnippet withChannelId(String channelId) {
        this.channelId = channelId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("defaultLanguage")
    public String defaultLanguage;
    public ChannelSectionSnippet withDefaultLanguage(String defaultLanguage) {
        this.defaultLanguage = defaultLanguage;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("localized")
    public ChannelSectionLocalization localized;
    public ChannelSectionSnippet withLocalized(ChannelSectionLocalization localized) {
        this.localized = localized;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("position")
    public Long position;
    public ChannelSectionSnippet withPosition(Long position) {
        this.position = position;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("style")
    public ChannelSectionSnippetStyleEnum style;
    public ChannelSectionSnippet withStyle(ChannelSectionSnippetStyleEnum style) {
        this.style = style;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("title")
    public String title;
    public ChannelSectionSnippet withTitle(String title) {
        this.title = title;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public ChannelSectionSnippetTypeEnum type;
    public ChannelSectionSnippet withType(ChannelSectionSnippetTypeEnum type) {
        this.type = type;
        return this;
    }
}

File: SDKs/googleapis.com/youtube/v3/python/src/sdk/models/shared/channelsectionsnippet.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import channelsectionlocalization as shared_channelsectionlocalization

class ChannelSectionSnippetStyleEnum(str, Enum):
    CHANNELSECTION_STYLE_UNSPECIFIED = "channelsectionStyleUnspecified"
    HORIZONTAL_ROW = "horizontalRow"
    VERTICAL_LIST = "verticalList"

class ChannelSectionSnippetTypeEnum(str, Enum):
    CHANNELSECTION_TYPE_UNDEFINED = "channelsectionTypeUndefined"
    SINGLE_PLAYLIST = "singlePlaylist"
    MULTIPLE_PLAYLISTS = "multiplePlaylists"
    POPULAR_UPLOADS = "popularUploads"
    RECENT_UPLOADS = "recentUploads"
    LIKES = "likes"
    ALL_PLAYLISTS = "allPlaylists"
    LIKED_PLAYLISTS = "likedPlaylists"
    RECENT_POSTS = "recentPosts"
    RECENT_ACTIVITY = "recentActivity"
    LIVE_EVENTS = "liveEvents"
    UPCOMING_EVENTS = "upcomingEvents"
    COMPLETED_EVENTS = "completedEvents"
    MULTIPLE_CHANNELS = "multipleChannels"
    POSTED_VIDEOS = "postedVideos"
    POSTED_PLAYLISTS = "postedPlaylists"
    SUBSCRIPTIONS = "subscriptions"


@dataclass_json
@dataclasses.dataclass
class ChannelSectionSnippet:
    r"""ChannelSectionSnippet
    Basic details about a channel section, including title, style and position.
    """
    
    channel_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('channelId') }})
    default_language: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('defaultLanguage') }})
    localized: Optional[shared_channelsectionlocalization.ChannelSectionLocalization] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('localized') }})
    position: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('position') }})
    style: Optional[ChannelSectionSnippetStyleEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('style') }})
    title: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('title') }})
    type: Optional[ChannelSectionSnippetTypeEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('type') }})
    
